lib/pam_pass_non_interactive.c

```diff
@@ -97,9 +98,8 @@ static int ni_conv (int num_msg,
 failed_conversation:
 	for (count=0; count < num_msg; count++) {
 		if (NULL != responses[count].resp) {
-			bzero(responses[count].resp,
-			      strlen(responses[count].resp));
-			free (responses[count].resp);
+			freezero(responses[count].resp,
+			         strlen(responses[count].resp));
```

Comment:
Alternatively, you may want to

```c
strzero(s);
free(s);
```

Reply:
I have my internal conflict, because it's simpler, but your approach is more portable.  I think simplicity tends to win, but it's your proposal, so whatever you prefer.  ;-)

Reply:
The combination of strzero and free is more often available in shadow than freezero. And no other example of freezero with strlen.

Thank you for the suggestion. I have adjusted the PR accordingly.

Reply:
=)